subsys/bluetooth/shell/bt.c

```diff
@@ -1245,6 +1247,289 @@ static int cmd_adv_info(const struct shell *shell, size_t argc, char *argv[])
+	selected_per_adv_sync = NULL;
+}
+
+static void per_adv_sync_recv_cb(
+	struct bt_le_per_adv_sync *sync,
+	const struct bt_le_per_adv_sync_recv_info *info,
+	struct net_buf_simple *buf)
+{
+	char le_addr[BT_ADDR_LE_STR_LEN];
+
+	if (selected_per_adv_sync != sync) {
+		shell_error(ctx_shell, "Invalid per adv sync object");
+		return;
+	}
+
+	bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+	shell_print(ctx_shell, "PER_ADV_SYNC[%u]: [DEVICE]: %s, tx_power %i, "
+		    "RSSI %i, CTE %u, data length %u",
+		    bt_le_per_adv_sync_get_index(sync), le_addr, info->tx_power,
+		    info->rssi, info->cte_type, buf->len);
+}
+
+static struct bt_le_per_adv_sync_cb per_adv_sync_cb = {
+	.synced = per_adv_sync_sync_cb,
+	.term = per_adv_sync_terminated_cb,
+	.recv = per_adv_sync_recv_cb
+};
+
+static int cmd_per_adv_sync_create(const struct shell *shell, size_t argc,
+				   char *argv[])
+{
+	int err;
+	struct bt_le_per_adv_sync_param create_params = { 0 };
+	uint32_t options = 0;
+	struct bt_le_per_adv_sync *free_per_adv_sync = NULL;
+
+	for (int i = 0; i < ARRAY_SIZE(per_adv_syncs); i++) {
+		if (per_adv_syncs[i] == NULL) {
+			free_per_adv_sync = per_adv_syncs[i];
+			break;
+		}
+	}
+
+	if (!free_per_adv_sync) {
+		shell_error(shell, "Cannot create more per adv syncs");
+		return -ENOEXEC;
+	}
+
+	err = bt_addr_le_from_str(argv[1], argv[2], &create_params.addr);
+	if (err) {
+		shell_error(shell, "Invalid peer address (err %d)", err);
+		return -ENOEXEC;
+	}
+
+	create_params.timeout = 1000; /* 10 seconds */
+	create_params.skip = 10;
+
+	if (argc > 3) {
+		create_params.sid = strtol(argv[3], NULL, 10);
+	}
+
+	if (argc > 4) {
+		create_params.skip = strtol(argv[4], NULL, 10);
+	}
+
+	if (argc > 5) {
+		create_params.timeout = strtol(argv[5], NULL, 10);
+	}
+
+	for (int i = 6; i < argc; i++) {
+		if (!strcmp(argv[i], "aoa")) {
+			options |= BT_LE_PER_ADV_SYNC_OPT_DONT_SYNC_AOA;
+		} else if (!strcmp(argv[i], "aod_1us")) {
+			options |= BT_LE_PER_ADV_SYNC_OPT_DONT_SYNC_AOD_1US;
+		} else if (!strcmp(argv[i], "aod_2us")) {
+			options |= BT_LE_PER_ADV_SYNC_OPT_DONT_SYNC_AOD_2US;
+		} else if (!strcmp(argv[i], "only_cte")) {
+			options |=
+				BT_LE_PER_ADV_SYNC_OPT_SYNC_ONLY_CONST_TONE_EXT;
+		}
+
+		/* TODO: add support to parse using the per adv list */
+	}
+
+	create_params.options = options;
+
+	err = bt_le_per_adv_sync_create(&create_params, &per_adv_sync_cb,
+					&free_per_adv_sync);
+	if (err) {
+		shell_error(shell, "Per adv sync failed (%d)", err);
+	} else {
+		shell_print(shell, "Per adv sync pending");
+	}
+
+	return 0;
+}
+
+static int cmd_per_adv_sync_delete(const struct shell *shell, size_t argc,
+				   char *argv[])
+{
+	int err;
+	int index = 0;
+	struct bt_le_per_adv_sync **per_adv_sync = NULL;
+
+	if (argc > 1) {
+		index = strtol(argv[1], NULL, 10);
+	}
+
+	for (int i = 0; i < ARRAY_SIZE(per_adv_syncs); i++) {
+		if (bt_le_per_adv_sync_get_index(per_adv_syncs[i]) == index) {
```

Comment:
You could use the index as the index into per_adv_sync table. Don't need the for loop, here and in create.

Reply:
The loop in create doesn't use an index, but the loop here can be replaced with your suggestion.